tests/resources/generated/ValidApiArrayOfGuid.php

```diff
@@ -0,0 +1,184 @@
+<?php
+
+declare(strict_types=1);
+
+namespace ArrayType;
+
+use InvalidArgumentException;
+use WsdlToPhp\PackageBase\AbstractStructArrayBase;
+
+/**
+ * This class stands for ArrayOfGuid ArrayType
+ * @package Api
+ * @subpackage Arrays
+ * @release 1.1.0
+ */
+class ApiArrayOfGuid extends AbstractStructArrayBase
+{
+    /**
+     * The string
+     * Meta information extracted from the WSDL
+     * - base: xsd:string
+     * - maxOccurs: unbounded
+     * - minOccurs: 0
+     * - pattern: [\da-fA-F]{8}-[\da-fA-F]{4}-[\da-fA-F]{4}-[\da-fA-F]{4}-[\da-fA-F]{12}
+     * @var string[]
+     */
+    protected ?array $string = null;
+    /**
+     * Constructor method for ArrayOfGuid
+     * @uses ApiArrayOfGuid::setString()
+     * @param string[] $string
+     */
+    public function __construct(?array $string = null)
+    {
+        $this
+            ->setString($string);
+    }
+    /**
+     * Get string value
+     * @return string[]
+     */
+    public function getString(): ?array
+    {
+        return $this->string;
+    }
+    /**
+     * This method is responsible for validating the values passed to the setString method
+     * This method is willingly generated in order to preserve the one-line inline validation within the setString method
+     * @param array $values
+     * @return string A non-empty message if the values does not match the validation rules
+     */
+    public static function validateStringForArrayConstraintsFromSetString(?array $values = []): string
+    {
+        if (!is_array($values)) {
+            return '';
+        }
+        $message = '';
+        $invalidValues = [];
+        foreach ($values as $arrayOfGuidStringItem) {
+            // validation for constraint: itemType
+            if (!is_string($arrayOfGuidStringItem)) {
+                $invalidValues[] = is_object($arrayOfGuidStringItem) ? get_class($arrayOfGuidStringItem) : sprintf('%s(%s)', gettype($arrayOfGuidStringItem), var_export($arrayOfGuidStringItem, true));
+            }
+        }
+        if (!empty($invalidValues)) {
+            $message = sprintf('The string property can only contain items of type string, %s given', is_object($invalidValues) ? get_class($invalidValues) : (is_array($invalidValues) ? implode(', ', $invalidValues) : gettype($invalidValues)));
+        }
+        unset($invalidValues);
+        
+        return $message;
+    }
+    public static function validateStringForPatternConstraintFromSetString($values): string
+    {
+        $message = '';
+        $invalidValues = [];
+        foreach ($values as $arrayOfGuidStringItem) {
+            // validation for constraint: pattern([\da-fA-F]{8}-[\da-fA-F]{4}-[\da-fA-F]{4}-[\da-fA-F]{4}-[\da-fA-F]{12})
+            if (!preg_match('/[\\da-fA-F]{8}-[\\da-fA-F]{4}-[\\da-fA-F]{4}-[\\da-fA-F]{4}-[\\da-fA-F]{12}/', $arrayOfGuidStringItem)) {
+                $invalidValues[] = var_export($arrayOfGuidStringItem, true);
+            }
```

Comment:
Would be better in the already generated method `validateStringForArrayConstraintsFromSetString`